packages/amplify_core/lib/types/plugin/amplify_base_plugin_interface.dart

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:meta/meta.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+/// Interface for all amplify plugins
+abstract class AmplifyBasePluginInterface extends PlatformInterface {

```

Comment:
Naming-wise, I'd suggest to drop "Base", both from the class name and the file name.

```suggestion
abstract class AmplifyPluginInterface extends PlatformInterface {
```

For reference, both [iOS](https://github.com/aws-amplify/amplify-ios/blob/main/Amplify/Core/Plugin/Plugin.swift#L11) and [Android](https://github.com/aws-amplify/amplify-android/blob/main/core/src/main/java/com/amplifyframework/core/plugin/Plugin.java#L47) call this just `Plugin`.

If my reading is correct, Amplify JS doesn't actually have any common base type for plugins? A "pluggable" is just something that can be added to an individual category, by convention?

Reply:
Refactoring in the branch.